board.es6

```diff
@@ -308,70 +310,62 @@ class Board {
+
     }
-    this.movementNotation.push(notation)
+
+    this.movementNotation.push( moveObject.notation() + (epNotation || "") + notationSuffix)
   }
 
-  _hypotheticallyMovePiece( moveObject ){
-    // there's a lot of space between _officiallyMovePiece and hypothetical. eg  not recording any data on hypothetical moves
+  _hypotheticallyMovePiece( moveObject ){ // ONLY USE THIS TO SEE IF A MOVE WOULD RESULT IN MATE. there's a lot of space between _officiallyMovePiece and hypothetical. eg  not recording any data on hypothetical moves
     let startPosition = moveObject.startPosition,
       endPosition = moveObject.endPosition,
-      additionalActions = moveObject.additionalActions,
-      pieceObject = this.pieceObject(startPosition);
+      additionalActions = moveObject.additionalActions;
+      let pieceObject = this.pieceObject(startPosition);
     this._emptify(startPosition)
     this._placePiece({ position: endPosition, pieceObject: pieceObject })
-    if( additionalActions ){ additionalActions.call(this, {position: startPosition} ) }
+    if( additionalActions ){ additionalActions.call(this, startPosition) }
   }
 
   _officiallyMovePiece( moveObject ){
-    if( !MoveObject.prototype.isPrototypeOf( moveObject ) ){ throw new Error("missing params in movePiece") }
+    // if( !MoveObject.prototype.isPrototypeOf( moveObject ) ){ throw new Error("missing params in movePiece") }
     let startPosition = moveObject.startPosition,
       endPosition = moveObject.endPosition,
       additionalActions = moveObject.additionalActions,
       pieceObject = this.pieceObject(startPosition);
-    this._storeCurrentLayoutAsPrevious()
-    this._emptify(startPosition)
-    moveObject.captureNotation = this._capture(endPosition);
-    this._placePiece({ position: endPosition, pieceObject: pieceObject })
-    if( additionalActions ){ moveObject.captureNotation = additionalActions.call(this, {position: startPosition} ) }
-    Rules.pawnPromotionQuery({board: this, moveObject: moveObject} );
-		Rules.checkmateQuery({board: this, moveObject: moveObject})
-    if( !this.gameOver ){
-      var otherTeam = this.teamNotMoving(),
-        otherTeamsKingPosition = this._kingPosition(otherTeam);
-      Rules.checkQuery( {startPosition: otherTeamsKingPosition, endPosition: otherTeamsKingPosition, board: this, moveObject: moveObject} )
-      Rules.stalemateQuery({board: this, moveObject: moveObject});
-    }
-    this._recordNotationFrom(moveObject)
-    if( !this.gameOver ){ this._nextTurn() }
-  }
 
-  _storeCurrentLayoutAsPrevious(){
-    let layOutCopy = Board._deepCopy( this.layOut );
-    this.previousLayouts.push(layOutCopy)
-  }
 
-  _capture(position){
-    let captureNotation = ""
-    if( !this.positionEmpty(position) ){
-      let pieceObject = this.layOut[position];
-      this.capturedPieces.push(pieceObject)
-      this._emptify(position)
-      captureNotation = "x"
+    let stringyLayOut = JSON.stringify(this.layOut)
```

Comment:
Hmm, what's going on here? The fact that you're munging the string layout seems a bit suspect.